cmd/kpod/ps.go

```diff
@@ -280,61 +391,86 @@ func getContainers(containers []*libkpod.ContainerData, opts psOptions) []*libkp
 }
 
 // getTemplateOutput returns the modified container information
-func getTemplateOutput(containers []*libkpod.ContainerData, opts psOptions) (psOutput []psTemplateParams) {
+func getTemplateOutput(containers []*libpod.Container, opts psOptions) ([]psTemplateParams, error) {
+	var psOutput []psTemplateParams
 	var status string
 	for _, ctr := range containers {
-		ctrID := ctr.ID
-		runningFor := units.HumanDuration(time.Since(ctr.State.Created))
+		ctrID := ctr.ID()
+		conConfig, err := ctr.GetConfig()
+		if err != nil {
+			return psOutput, errors.Wrapf(err, "unable to obtain container config")
+		}
+		conState, err := ctr.State()
+		if err != nil {
+			return psOutput, errors.Wrapf(err, "unable to container state")
+		}
+		exitCode, err := ctr.ExitCode()
+		if err != nil {
+			return psOutput, errors.Wrapf(err, "unable to obtain container exit code")
+		}
+		pid, err := ctr.PID()
+		if err != nil {
+			return psOutput, errors.Wrapf(err, "unable to obtain container pid")
+		}
+		runningFor := units.HumanDuration(time.Since(conConfig.CreatedTime))
 		createdAt := runningFor + " ago"
-		command := getStrFromSquareBrackets(ctr.ImageCreatedBy)
-		imageName := ctr.FromImage
-		mounts := getMounts(ctr.Mounts, opts.noTrunc)
-		ports := getPorts(ctr.Config.ExposedPorts)
-		size := units.HumanSize(float64(ctr.SizeRootFs))
-		labels := getLabels(ctr.Labels)
-
-		ns := getNamespaces(ctr.State.Pid)
-
-		switch ctr.State.Status {
-		case oci.ContainerStateStopped:
-			status = "Exited (" + strconv.FormatInt(int64(ctr.State.ExitCode), 10) + ") " + runningFor + " ago"
-		case oci.ContainerStateRunning:
+		imageName := conConfig.RootfsImageName
+
+		// TODO We currently dont have the ability to get many of
+		// TODO these data items.  Uncomment as progress is made
```

Comment:
nit of a nit, I'd remove TODO here if you make other changes to the file.

Reply:
done